sqlglot/expressions.py

```diff
@@ -1117,3 +1570,141 @@ def _all_functions():
 
 
 ALL_FUNCTIONS = _all_functions()
+
+
+def _maybe_parse(
+    code_or_expression,
+    *,
+    expression_type=None,
+    dialect=None,
+    prefix=None,
+    parser_opts=None,
+):
+    if isinstance(code_or_expression, Expression):
+        return code_or_expression
+
+    parser_opts = parser_opts or {}
+    code = str(code_or_expression)
+    if prefix:
+        code = f"{prefix} {code}"
+
+    from sqlglot.dialects import Dialect
+
+    dialect = Dialect.get_or_raise(dialect)()
+
+    if expression_type:
+        result = dialect.parse_into(expression_type, code, **parser_opts)
+    else:
+        result = dialect.parse(code, **parser_opts)
+
+    if not result:
+        return None
+    return result[0]
+
+
+T = TypeVar("T", bound=Expression)
+
+
+def _maybe_copy(instance: T, copy=True) -> T:
```

Comment:
_maybe_copy is an unnecessary method, and kind of confusing because i had to look it up, should we just inline these?

Reply:
x = x.copy() if copy else x

Reply:
i guess you call this a ton, maybe it's fine

Reply:
Yeah, there's a handful of clunky helper functions here. But I created them because they're needed a few times. 🤷